docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -2015,188 +1806,211 @@ If a transaction query is created with a templateId specified as
+upgrade, and thus the full transaction, fails. Contracts in this
+state can then only be used by explicitly choosing the older version of
+the contract in your transaction.
 
-  # Path to the dar you want to pass as package preference.
-  PACKAGE_DAR=path/to/main/dar.dar
-  # Path to the dar containing the Daml script for which you want to pass the package-id
-  SCRIPT_DAR=path/to/script/dar.dar
-  # Extract the package-id of PACKAGE_DAR's main package.
-  daml damlc inspect-dar ${PACKAGE_DAR} --json | jq '.main_package_id' > ./package-id-script-input.json
-  # replace --ide-ledger with --ledger-host and --ledger-port for deployed Canton
-  daml script --dar ${SCRIPT_DAR} --script-name Main:main --ide-ledger --input-file ./package-id-script-input.json
+.. _upgrade_package_naming:
 
-Following this, your script would look like
+Breaking Changes via Explicit Package Version
+---------------------------------------------
 
-.. code:: daml
+To make a breaking change to your package that
+is not upgrade compatible, you can change the name of your package to indicate a
+breaking version bump. To enable this, we recommend that your package
+name contains a version marker for when a breaking change occurs.
 
-  module Main where
+For example, for your first iteration of a package, you would name it
+``main-v1``, starting with package version ``1.0.0``. In this case, the ``v1``
+is part of the *package name*, not the package version. You could publish
+upgrade-compatible versions by changing the ``version:`` field from ``1.0.0`` to
+``2.0.0`` to ``3.0.0``. These versions would all be upgrade-compatible with
+one another:
 
-  import Daml.Script
+.. code:: text
 
-  main : Text -> Script ()
-  main rawPkgId = do
-    let pkgId = PackageId rawPkgId
-    alice <- allocateParty "alice"
-    -- Commands omitted for brevity
-    let submitOptions = actAs alice <> packagePreference [pkgId]
-    submitOptions `submitWithOptions` createCmd ...
+  main-v1-1.0.0
+  main-v1-2.0.0
+  main-v1-3.0.0
 
-Daml Studio support
--------------------
+Note how the ``v1`` in all three packages remains stable - this means the
+package name has not changed, and ensures that these three packages and their
+datatypes are considered by the runtime and the type checker to be upgradeable.
 
-Daml Studio runs a reference model of Canton called the IDE Ledger. This
-ledger has been updated to support the relevant parts of the Smart Contract
-Upgrades feature.
+When you want to make a breaking change, you would publish a new version of the
+package with package name ``main-v2``. Because this package would have a
+different package name from those with ``main-v1``, it would not be typechecked
+against those packages and its datatypes would not automatically be converted.
+You would need to manually migrate values from ``main-v1`` packages to
+``main-v2`` -- existing downtime upgrade techniques are listed :ref:`here <upgrades-index>`.
```

Comment:
```suggestion
To make a breaking change, publish a new package version with package name ``main-v2``. Because this package has a
different package name from those with ``main-v1``, it is not typechecked
against those packages and its datatypes are not automatically converted.
You would need to manually migrate values from ``main-v1`` packages to
``main-v2`` -- existing downtime upgrade techniques are listed :ref:`here <upgrades-index>`.
```